File: js-pr/56/javascript/code-quality-readability-maintainability/data_validator_07.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function validate(data) {
    if (!data) return false;
    if (typeof data !== 'object') return false;
    if (!data.name) return false;
    if (!data.age) return false;
    if (typeof data.name !== 'string') return false;
    if (typeof data.age !== 'number') return false;
    if (data.name.length < 2) return false;
    if (data.age < 0) return false;
    if (data.age > 150) return false;
    return true;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/data_validator_07.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def validate(data):
    if not data: return False
    if not isinstance(data, dict): return False
    if 'name' not in data: return False
    if 'age' not in data: return False
    if not isinstance(data['name'], str): return False
    if not isinstance(data['age'], int): return False
    if len(data['name']) < 2: return False
    if data['age'] < 0: return False
    if data['age'] > 150: return False
    return True
# {/fact}